starknet_py/utils/typed_data/functions.py

```diff
@@ -0,0 +1,89 @@
+from typing import List, Union
```

Comment:
For me this module is too complicated to comprehend. I think you should introduce more abstraction to make it easier to comprehend.

Also, variables like `typed_data` are getting passed around bym multiple methods. If they are really necessary in every one of them, this should be converted into a class with necessary values assigned to `self`. But again, I think there should be more abstraction.